vm/src/stdlib/pystruct.rs

```diff
@@ -88,13 +88,35 @@ where
     I: Sized + Iterator<Item = char>,
 {
     let mut codes = vec![];
-    for c in chars {
-        match c {
-            'b' | 'B' | 'h' | 'H' | 'i' | 'I' | 'l' | 'L' | 'q' | 'Q' | 'f' | 'd' => {
-                codes.push(FormatCode { code: c })
+    while chars.peek().is_some() {
+        // determine repeat operator:
+        let repeat = match chars.peek() {
+            Some('0'..='9') => {
+                let mut repeat = 0;
+                while let Some('0'..='9') = chars.peek() {
+                    if let Some(c) = chars.next() {
+                        let current_digit = c.to_digit(10).unwrap();
+                        repeat = repeat * 10 + current_digit;
+                    }
+                }
+                Some(repeat)
             }
-            c => {
-                return Err(format!("Illegal format code {:?}", c));
+            _ => None,
+        };
+
+        // determine format char:
+        if let Some(c) = chars.next() {
```

Comment:
The else arm of this if should return an error.

Reply:
Does this work?
```rust
if let Some('b' | 'B' | ...) = chars.next() {
    ...
} else {
    // massive error
}
```

Reply:
There are syntax errors in your suggestion. But I got inspiration from that.